src/tunacode/tools/decorators.py

```
@@ -216,6 +226,23 @@ async def execute(
+def _validate_strict_bound_arguments(
+    sig: inspect.Signature,
+    bound: inspect.BoundArguments,
+) -> None:
+    for param_name, value in list(bound.arguments.items()):
+        param = sig.parameters.get(param_name)
+        if param is None or param.annotation is inspect.Parameter.empty:
+            continue
+        try:
+            validated = TypeAdapter(param.annotation).validate_python(value, strict=True)
+        except ValidationError as exc:
+            raise TypeError(f"parameter '{param_name}' failed strict validation: {exc}") from exc
```

Ruff (0.15.6)

[warning] Avoid specifying long messages outside the exception class

(TRY003)